component/attr_cache/attr_cache.go

```diff
@@ -868,17 +871,22 @@ func (ac *AttrCache) GetAttr(options internal.GetAttrOptions) (*internal.ObjAttr
 		// Retrieved attributes so cache them
 		// TODO: bug: when cacheDirs is true, the cache limit will cause some directories to be double-listed
 		// TODO: shouldn't this be an LRU? This sure looks like the opposite...
-		if len(ac.cacheMap) < ac.maxFiles {
-			ac.cacheMap[truncatedPath] = newAttrCacheItem(pathAttr, true, time.Now())
+		if pathAttr == nil {
+			ac.cacheMap.insert(&internal.ObjAttr{Path: options.Name}, true, time.Now())
 		} else {
-			log.Debug("AttrCache::GetAttr : %s skipping adding to attribute cache because it is full", options.Name)
+			ac.cacheMap.insert(pathAttr, true, time.Now())
 		}
+
 		if ac.cacheDirs {
 			ac.markAncestorsInCloud(getParentDir(options.Name), time.Now())
 		}
 	} else if err == syscall.ENOENT {
 		// Path does not exist so cache a no-entry item
-		ac.cacheMap[truncatedPath] = newAttrCacheItem(&internal.ObjAttr{}, false, time.Now())
+		// this insert should involve the key of the cacheMap item being the truncatedPath. Do we have insert some how know the truncated path to use as the key? or do we do a key literal insert right here? the latter is mesyier for cacheMap access scope.
+
+		//ac.cacheMap.insert(&internal.ObjAttr{}, false, time.Now())
```

Comment:
The two lines below are incorrect and need to be removed.
We need to insert a cache item with a new `ObjAttr` to record that the requested file/folder does not exist.
The `ObjAttr` needs to have a valid path for `insert()` to work, so we need to create an `ObjAttr` with  `truncatedPath`. You should be able to use either `CreateObjAttr()` or `CreateObjAttrDir()` and it shouldn't matter. So since you only have the path (and not the size, etc), you might as well use that one.